generate/util/util.go

```diff
@@ -0,0 +1,235 @@
+package util
+
+import (
+	"fmt"
+	"reflect"
+	"time"
+	"unsafe"
+
+	"github.com/segmentio/objconv"
+	"github.com/segmentio/objconv/objutil"
+)
+
+// SkipValue properly skips a value in a parser based on the type
+func SkipValue(p objconv.Parser) (err error) {
+	var typ objconv.Type
+
+	typ, err = p.ParseType()
+	if err != nil {
+		return err
+	}
+
+	switch typ {
+	case objconv.Nil:
+		return p.ParseNil()
+	case objconv.Bool:
+		_, err = p.ParseBool()
+	case objconv.Int:
+		_, err = p.ParseInt()
+	case objconv.Uint:
+		_, err = p.ParseUint()
+	case objconv.Float:
+		_, err = p.ParseFloat()
+	case objconv.String:
+		_, err = p.ParseString()
+	case objconv.Bytes:
+		_, err = p.ParseBytes()
+	case objconv.Time:
+		_, err = p.ParseTime()
+	case objconv.Duration:
+		_, err = p.ParseDuration()
+	case objconv.Error:
+		_, err = p.ParseError()
+	case objconv.Array:
+		var count int
+		count, err = p.ParseArrayBegin()
+		if err != nil {
+			return
+		}
+		var i int
+		for i = 0; count < 0 || i < count; i++ {
+			if count < 0 || i != 0 {
+				err = p.ParseArrayNext(i)
+				if err == objconv.End {
+					break
+				}
+				if err != nil {
+					return
+				}
+			}
+			if err = SkipValue(p); err != nil {
+				return
+			}
+		}
+		err = p.ParseArrayEnd(i)
+	case objconv.Map:
+		var count int
+		count, err = p.ParseMapBegin()
+		if err != nil {
+			return err
+		}
+		var i int
+		for i = 0; count < 0 || i < count; i++ {
+			if count < 0 || i != 0 {
+				err = p.ParseMapNext(i)
+				if err == objconv.End {
+					break
+				}
+				if err != nil {
+					return
+				}
+			}
+			if err = SkipValue(p); err != nil {
+				return
+			}
+			if err = p.ParseMapValue(i); err != nil {
+				return
+			}
+			if err = SkipValue(p); err != nil {
+				return
+			}
+		}
+		err = p.ParseMapEnd(i)
+	default:
+		err = fmt.Errorf("Unknown type: %v", typ)
+	}
+	return
+}
+
+// ParseInt parses a value into an int64
+func ParseInt(p objconv.Parser) (int64, error) {
+	typ, err := p.ParseType()
+	if err != nil {
+		return 0, err
+	}
+	var v int64
+	switch typ {
+	case objconv.Int:
+		v, err = p.ParseInt()
+	case objconv.Uint:
+		var uv uint64
+		uv, err = p.ParseUint()
+		if uv > objutil.Int64Max {
+			err = fmt.Errorf("objconv: %d does not fit in int64", uv)
+		}
+		v = int64(uv)
+	case objconv.Float:
+		var fv float64
+		fv, err = p.ParseFloat()
```

Comment:
Should we check that the fractional part is zero as well?

This could help => https://golang.org/pkg/math/#Modf

Reply:
Ah that's what the double cast was for: `float64(int64(fv)) != fv`. this also catches very large numbers that get clobbered on a cast to int64